students/larry_mburu/Lesson06/mailroom.py

```diff
@@ -0,0 +1,213 @@
+    #!/usr/bin/env python3
+
+import sys
+import os
+from operator import itemgetter
+
+donations_per_individual = { 
+    'William Gates III' : [50000.0, 500.12], 
+    'Mark Zuckerberg' : [20000.0, 8500.99], 
+    'Jeff Bezo': [100000.0, 40000.0, 700.99], 
+    'Paul Allen': [200000.0, 1440.0, 300.00], 
+    'Bill Gates': [30000.0, 70000.0, 450.0]
+}
+
+def display_menu():
+    """
+    Display menu option to user
+    """
+    print("Welcome to Mail Automator (MA)!")
+    print()
+    print("[1] Send a Thank You!")
+    print("[2] Create a report!")
+    print("[3] Send letters to all donors")
+    print("[4] Quit!")
+    print()
+
+def send_thank_you_letters(): 
+    """
+    Function, sends a thank you letter to all donor's in the donor DB list,
+    by writing thank you letters to the local filesystem.
+    """
+    print('[+] Generating thank you letters..')
+
+    for donor_name in donations_per_individual:  
+        #split and join,to prevent file names with spaces.
+        with open('_'.join(donor_name.split()) + '.txt', 'w') as fobj: 
+            fobj.write(get_letter_text(donor_name))
+
+    print('[+] Letter generation complete.')
+    print('[+] Letters are in your current working directory')
+
+def get_letter_text(donor_name):
+    """
+    returns formatted string, for content to be written
+    in each file.
+    """
+
+    # -1, the latest donation in the donor "list"
+    email_template = f"""Dear {donor_name},
+    Thank you for your kind donation of ${donations_per_individual[donor_name][-1]}. 
+        
+    It will be put to very good use. 
+        
+    Sincerely,
+
+    - The Cloud Squad"""
+    
+    return email_template
+
+def send_thank_you_email(name, amount): 
+    """
+    composes an email to a selected donor on the donor's list 
+    """
+    thank_you_note = f"Thank you {name} for your generous donation of $ {amount:.2f} dollars"
+    print(thank_you_note)
+    return thank_you_note
+
+def prompt_for_donor_name(): 
+    """
+    Function, asks user for a donar name. The user can opt to
+    list all the donors in the donor DB list. 
+    """
+    donor_name = input("Enter donor's full name or 'list' to list donors: ")
+
+    while donor_name == 'list': 
+        print(get_donor_list(donor_name))
+        donor_name = input("Enter donor's full name or 'list' to list donors: ")
+    
+    add_donar_to_list(donor_name)
+    
+def get_donor_list(donor_name): 
+    """
+    Gets a current listing of donors 
+    Param: donor_name: if list, return a list of donors
+    """
+    if donor_name == 'list': 
+        return list(donations_per_individual) #list of donors
```

Comment:
why is this function taking in param? it's only purpose should be to return donor names